### PR TITLE
fix: remove redundant reexport from external modules

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
I think it is not related to https://github.com/rolldown/rolldown/pull/1773?, since main branch also failed, #1773 just keep `export * from 'fs'` as is in entry module.

**Comment by hyf0** [comment]
@IWANABETHATGUY It might be related. There are conditions that using `is_user_defined_entry` might fall into this problem too. Let's have a chat afternoon.

File: crates/rolldown/tests/fixtures/semantic/export_star_from_external_as_shared_entries_cjs/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+expression: snapshot
+input_file: crates/rolldown/tests/fixtures/semantic/export_star_from_external_as_shared_entries_cjs
+---
+# Assets
+
+## entry.cjs
```
Comment (hyf0) [inline crates/rolldown/tests/fixtures/semantic/export_star_from_external_as_shared_entries_cjs/artifacts.snap]: format cjs behavior is still wrong.

File: crates/rolldown/tests/esbuild/import_star/re_export_star_external_es6/artifacts.snap [hunk 2]
```diff
@@ -10,6 +10,8 @@ input_file: crates/rolldown/tests/esbuild/import_star/re_export_star_external_es
 ```js
 import "foo";
 
+export * from "foo"
+
 
 //#region entry.js
 var entry_ns = {};
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/import_star/re_export_star_external_es6/artifacts.snap]: Was this more like a workaround or a correct fix? the `foo` is exported twice?

Comment (hyf0) [reply crates/rolldown/tests/esbuild/import_star/re_export_star_external_es6/artifacts.snap]: Correct fix. But the current code of rendering imports is kind of over abstraction. Don't want to deal with it in this timing.

**Pull request closed without merging**